src/agent/rustjail/src/cgroups/fs/mod.rs

```diff
@@ -27,1120 +43,848 @@ macro_rules! sl {
-        Ok(v) => {
-            if !v.trim().is_empty() {
-                info!(sl!(), "value: \"{}\"", v.as_str().trim());
-                return write_file(dir, file, v);
-            } else {
-                copy_parent(parent, file)?;
-                return copy_parent(dir, file);
+            let realtime_runtime = cpu.realtime_runtime.unwrap_or(0);
+            if realtime_runtime != 0 {
+                cpu_controller.set_rt_runtime(realtime_runtime);
+            }
+            let realtime_period = cpu.realtime_period.unwrap_or(0);
+            if realtime_period != 0 {
+                cpu_controller.set_rt_period_us(realtime_period);
             }
         }
-        Err(ref e) if io_error_kind_eq(e, std::io::ErrorKind::NotFound) => {
-            copy_parent(parent, file)?;
-            return copy_parent(dir, file);
-        }
-        Err(e) => return Err(e.into()),
-    }
-}
 
-fn write_nonzero(dir: &str, file: &str, v: i128) -> Result<()> {
-    if v != 0 {
-        write_file(dir, file, v.to_string())?;
-    }
+        if r.memory.is_some() {
+            info!(sl!(), "cgroup manager set memory !!!");
+            let mem_controller: &MemController = cg.controller_of().unwrap();
+            let memory = r.memory.as_ref().unwrap();
 
-    Ok(())
-}
+            if !update {
+                // initialize kmem limits for accounting
+                mem_controller.set_kmem_limit(1);
+                mem_controller.set_kmem_limit(-1);
+            }
 
-fn try_write_nonzero(dir: &str, file: &str, v: i128) -> Result<()> {
-    match write_nonzero(dir, file, v) {
-        Ok(_) => Ok(()),
-        Err(ref e) if io_error_kind_eq(e, std::io::ErrorKind::PermissionDenied) => Ok(()),
-        Err(e) => Err(anyhow!(e)),
-    }
-}
+            let limit = memory.limit.unwrap_or(0);
+            if limit != 0 {
+                mem_controller.set_limit(limit);
+            }
 
-fn remove(dir: &str) -> Result<()> {
-    fs::remove_dir_all(dir)?;
-    Ok(())
-}
+            let reservation = memory.reservation.unwrap_or(0);
+            if reservation != 0 {
+                mem_controller.set_soft_limit(reservation);
+            }
 
-fn apply(dir: &str, pid: pid_t) -> Result<()> {
-    write_file(dir, CGROUP_PROCS, pid)?;
-    Ok(())
-}
+            let swap = memory.swap.unwrap_or(0);
+            if cg.v2() {
+                let swap = convert_memory_swap_to_v2_value(swap, limit)?;
+            }
+            mem_controller.set_memswap_limit(swap);
 
-fn try_write_file<T: ToString>(dir: &str, file: &str, v: T) -> Result<()> {
-    match write_file(dir, file, v) {
-        Err(e) => {
-            let err = Errno::last();
-            if err == Errno::EINVAL || err == Errno::ENODEV || err == Errno::ERANGE {
-                warn!(sl!(), "Invalid Arguments!");
-                return Ok(());
+            let kernel = memory.kernel.unwrap_or(0);
+            if kernel != 0 {
+                mem_controller.set_kmem_limit(kernel);
```

Comment:
I'm a bit curious why the compiler does not complain here. `set_kmem_limit` returns a `Result<()>`, so the compiler should demand that you use it. Why not?

In other words, I would have expected this to be either some error handling or at the very minimum end with a `?`

Reply:
Yes, result should be handled.